sklearn/linear_model/ridge.py

```diff
@@ -203,6 +195,23 @@ def _deprecate_dense_cholesky(solver):
     return solver
 
 
+def _rescale_data(X, y, sample_weight):
+    """Rescale data so as to support sample_weight"""
+    n_samples = X.shape[0]
+    sample_weight = sample_weight * np.ones(n_samples)
```

Comment:
There is a branch at https://github.com/mblondel/scikit-learn/blob/sample_weights/sklearn/linear_model/ridge.py#L320 I was just assuming that `_rescale_data` is not called if `sample_weight` is None.